lib/orogen/gen/project.rb

```diff
@@ -676,14 +676,21 @@ def generate
                 Generation.save_automatic "orogen-project-#{name}.pc.in", pc
 
 		if !self_tasks.empty?
+                    
 		    self_tasks.each { |t| t.generate }
 
 		    deployer = Generation.render_template "tasks", "DeployerComponent.cpp", binding
 		    Generation.save_automatic "tasks", "DeployerComponent.cpp", deployer
 		    pc = Generation.render_template "tasks", "tasks.pc", binding
 		    Generation.save_automatic "tasks", "#{name}-tasks.pc.in", pc
+
+                    self_tasks.each do |t| 
+                        proxies = Orocos::Generation::CppProxyGeneration.new(component, t)
```

Comment:
better name for 'component' is 'self' here. 'component' is exported for the benefit of the code generation templates.